### PR TITLE
[MM-18324] Migrate tests from "app/plugin_deadlock_test.go" to use testify

### PR DESCRIPTION
**Pull request by phillipahereza**
#### Summary
Migrates tests from `app/plugin_deadlock_test.go` to use testify

#### Ticket Link
[MM-18324](https://mattermost.atlassian.net/browse/MM-18324)

Fixes https://github.com/mattermost/mattermost-server/issues/12062

### CODE REVIEW

File: app/plugin_deadlock_test.go [hunk 1]
```diff
@@ -4,6 +4,7 @@
 package app
 
 import (
+	"github.com/stretchr/testify/require"
```
Comment (lieut-data) [inline app/plugin_deadlock_test.go]: Indentation here seems off, or is it just GitHub?

Comment (phillipahereza) [reply app/plugin_deadlock_test.go]: @lieut-data probably GitHub, `make gofmt` makes no changes on my side

Comment (ali-farooq0) [reply app/plugin_deadlock_test.go]: @lieut-data it's `Refined Github` that's causing the indentation issue. I'm seeing the same. Probably an issue on their end.

**Comment by mgdelacroix** [comment]
No QA is needed as this PR only modifies the tests code.

**Pull request merged**